presto-tablestore/src/main/java/com/facebook/presto/tablestore/TablestoreValueUtil.java

```diff
@@ -0,0 +1,221 @@
+                }
+                if (x != null) {
+                    if (toPrimaryKeyNotColumnValue) {
+                        return (T) PrimaryKeyValue.fromLong(x);
+                    }
+                    else {
+                        return (T) ColumnValue.fromLong(x);
+                    }
+                }
+                acceptedTypes = new Class[] {String.class, Integer.class, Long.class, Short.class, Byte.class, Character.class};
+            }
+            else if (type instanceof BooleanType) {
+                Boolean x = null;
+                if (value instanceof Boolean) {
+                    x = (Boolean) value;
+                }
+                else if (value instanceof String) {
+                    x = Boolean.parseBoolean((String) value);
+                }
+                else if (value instanceof Number) {
+                    x = ((Number) value).longValue() != 0;
+                }
+                if (x != null && !toPrimaryKeyNotColumnValue) {
+                    return (T) ColumnValue.fromBoolean(x);
+                }
+                acceptedTypes = new Class[] {String.class, Number.class, Boolean.class};
+            }
+            else if (type instanceof DoubleType || type instanceof RealType) {
+                Double x = null;
+                if (value instanceof Double) {
+                    x = (Double) value;
+                }
+                else if (value instanceof String) {
+                    x = Double.parseDouble((String) value);
+                }
+                else if (value instanceof Number) {
+                    x = ((Number) value).doubleValue();
+                }
+                if (x != null && !toPrimaryKeyNotColumnValue) {
+                    return (T) ColumnValue.fromDouble(x);
+                }
+                acceptedTypes = new Class[] {String.class, Number.class, Double.class};
+            }
+            else if (type instanceof VarbinaryType) {
+                byte[] x = null;
+                if (value instanceof byte[]) {
+                    x = (byte[]) value;
+                }
+                else if (value instanceof String) {
+                    x = ((String) value).getBytes(UTF_8);
+                }
+                else if (value instanceof Slice) {
+                    x = ((Slice) value).getBytes();
+                }
+                if (x != null) {
+                    if (toPrimaryKeyNotColumnValue) {
+                        return (T) PrimaryKeyValue.fromBinary(x);
+                    }
+                    else {
+                        return (T) ColumnValue.fromBinary(x);
+                    }
+                }
+                acceptedTypes = new Class[] {String.class, byte[].class, Slice.class};
+            }
+        }
+        catch (Throwable e) {
+            String x = String.valueOf(value);
+            String str = "convert() failed for " + (toPrimaryKeyNotColumnValue ? "PrimaryKeyValue" : "ColumnValue") +
```

Comment:
is it better to use String.format? your call
Also we might need a more meaningful name instead of str

Reply:
done